Documentation/devel/distribution.md

```diff
@@ -0,0 +1,147 @@
+
+**NOTE:** I'm not sure if distribution is the best name for this concept or it should be called with other names like "image reference". So I'd like to hear your thoughts.
+
+A distribution represents the method to fetch an image starting from an input string. It doesn't mandates the image types. Usually a distribution can only provide one image type but some can provide different image types. For example:
+
+* Some docker registries also provides OCI images.
+* Rkt can internally implement the docker distribution to natively fetch a docker/OCI image from a registry or an ACI converted by docker2aci.
+
+Before this concept was introduced, rkt used the ImageType concept, the ImageType was mapped to the input image string format and internally covered multiple concepts like distribution, transport and image type (hidden since all are now appc ACIs).
+
+The distribution concept is used as the primary information in the different layers of rkt (fetching but also for references in a CAS/ref store).
+
+## Distribution types
+
+The distribution can be logically split in two types:
+
+* Name based distribution
+* Location based distribution
+
+Name based distribution can be considered the first class distribution type as it'll be the primarly used distribution type.
+Location based distribution can be thought as a sub-distribution for Name based distribution or a fast alternative (but not covering the Name based distribution feature) to directly fetch an image from a location.
```

Comment:
I think this sentence and the one above only add more confusion. Primary usage depends on specific usecases. Again, I'd speak about direct and indirect resource referencing.